### PR TITLE
Add new e2e test shopper cart calculate shipping

### LINKED ISSUE #29415: Write new e2e test: Shopper - Cart - Calculate shipping address
Write a new e2e test that covers the following flow:

- Shopper - Cart - Calculate shipping address

### PR DESCRIPTION
**Pull request by veljkho**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #29415

### How to test the changes in this Pull Request:

- npx wc-e2e test:e2e-dev ./tests/e2e/specs/front-end/test-cart-calculate-shipping.js

NOTE:
In addition to this test, I previously made [this test case](https://github.com/woocommerce/woocommerce/blob/trunk/tests/e2e/core-tests/specs/merchant/wp-admin-settings-shipping-zones.test.js) which also tests shopper's cart calculation, but this time I added new scenarios so we have the feature covered completely.

Also, I would mention that in this PR I have added a new component `deleteAllShippingZones` and updated other test to use it. I wanted to have it also in use for this test but realized that I don't need it for this test.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	it,
+	describe,
+	beforeAll,
+} = require( '@jest/globals' );
+
+const config = require( 'config' );
+const firstProductPrice = config.has( 'products.simple.price' ) ? config.get( 'products.simple.price' ) : '9.99';
+const secondProductPrice = '4.99';
+const fourProductPrice = firstProductPrice * 4;
+var twoProductsPrice = eval([firstProductPrice,secondProductPrice].join('+'));
```
Comment (zhongruige) [inline tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js:29]: I am a little hesitant to use `eval()`, even though just in tests due to the security risks associated with it. What are your thoughts on potentially casting to ints or something else here?

Comment (veljkho) [reply tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js:29]: I don't know what security risks you were referring to? I tried locally something but was an overcomplicated way. Could you please suggest something on how to improve it?

Comment (zhongruige) [reply tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js:29]: > I don't know what security risks you were referring to?

Oh, sorry, I was just referring to the general recommendation not to use it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#never_use_eval!

> tried locally something but was an overcomplicated way. 

It's very likely you tried these, but a couple of ideas I had include using the unary plus operator, so something like this:

```
var twoProductsPrice = +firstProductPrice + +secondProductPrice;
```

or

```
var twoProductsPrice = (+firstProductPrice) + (+secondProductPrice);
```

Alternatively, we could use the `parseFloat()` function:

```
var twoProductsPrice = parseFloat(firstProductPrice) + parseFloat(secondProductPrice);
```

We could even abstract the calculation out into a helper function and just call that instead so we aren't duplicating it line after line.

Let me know your thoughts, and if you think `eval()` should just be fine to use here too.

Comment (veljkho) [reply tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js]: Thanks @zhongruige !
I go with the second option, it's easy to understand and to read.

Comment (zhongruige) [reply tests/e2e/core-tests/specs/shopper/front-end-cart-calculate-shipping.test.js]: Sounds good, thanks @veljkho!

**Review by zhongruige — COMMENTED** [review]
The changes look good! I was able to run the tests both in headless and non-headless and they passed:

Screen Shot 2021-04-15 at 1 07 03 PM

Screen Shot 2021-04-15 at 1 07 52 PM

However when I run the entire test suite in headless/non-headless mode, the tests have been failing for me consistently, so I am curious if this is passing for you locally when running the full suite as well @veljkho . It's odd since it's all passed in the CI, so want to see if it's just some weirdness in mt environment.

Screen Shot 2021-04-15 at 1 17 32 PM

**Comment by veljkho** [comment]
Hey @zhongruige 
I didn't have that problem locally.. was running and passing. Could it be due to some leftover shipping zones? Try to manually delete any shipping zones.

**Comment by zhongruige** [comment]
> Hey @zhongruige
> I didn't have that problem locally.. was running and passing. Could it be due to some leftover shipping zones? Try to manually delete any shipping zones.

Thanks for double checking, I cleared everything out completely and set things up again (was having other weird issues and it seems fine for me now.

**Review by zhongruige — APPROVED** [review]

**Pull request merged**